conmon-rs/server/src/pty.rs

```diff
@@ -0,0 +1,222 @@
+//! Pseudo terminal implementation.
+
+use anyhow::{Context, Result};
+use getset::{CopyGetters, Setters};
+use log::{debug, error};
+use nix::{
+    fcntl::OFlag,
+    sys::stat::{self, Mode},
+    unistd,
+};
+use std::{
+    fs::OpenOptions,
+    io::{self, BufReader, ErrorKind, Read, Write},
+    os::unix::io::{IntoRawFd, RawFd},
+    str, thread,
+};
+
+#[derive(Clone, Copy, Debug, CopyGetters, Setters)]
+#[getset(get_copy, set)]
+pub struct Pty {
+    main_stdin: Stream,
+    main_stderr: Stream,
+    main_stdout: Stream,
+    worker_stdin: Stream,
+    worker_stderr: Stream,
+    worker_stdout: Stream,
+}
```

Comment:
do you have an idea how !terminal stream handling will work? if it will use this struct (as I see we're using these fields when dup'ing), maybe it should have a different name, as it won't actually be a pty

Reply:
My idea was to make it technically work in the same way as conmon is doing it today. Do you think there is an easier or better way to do it? (if so, let's sync on this topic)

I was thinking about a better name for `Pty`. Using `IOStreams` for now, maybe you have something better in mind.

Reply:
yeah IOStreams is good, and I +1 the idea of keeping in sync with old conmon